packages/react/src/events.ts

```diff
@@ -39,3 +45,14 @@ export function action<T = any>(name: T): ActionDataState<T> {
     name,
   }
 }
+
+export function withState<T extends Record<string, any>>(
+  state: { [K in keyof T]: StateDataState<T[K]> },
```

Comment:
this makes it so that arguments to your `fn` are exactly as the keys/values passed as `state`.

Reply:
nice. Should we do `Record<string, unknown>` so that TS will continue inferring?

Reply:
Surprisingly, it does infer it 100% perfect even though there's `any`!